scripts/GWA_TBX/PG04/PG04_WaterQualityWorkflow_SNAP_graphProcessor.py

```diff
@@ -52,363 +52,364 @@ def locate_file(pattern, root_path):
     for path, dirs, files in os.walk(root_path):
         for filename in fnmatch.filter(files, pattern):
             yield os.path.join(path, filename)
-        
-def create_graph(tempdir, subset):
+
+def create_graph(tempdir, subset, NN_PATH):
     with open(tempdir + "ProcessingGraph.xml", "w") as text_file:
```

Comment:
Sorry for the late answer. This was by design, as some parts of the graph are variable, depending on the sensor (OLCI or MERIS) This way, I don't have to have two scripts. From my perspective it is quite clean to have the xml in the script.

Reply:
Well, the recommended way in SNAP is to use parameters [like described here](https://senbox.atlassian.net/wiki/spaces/SNAP/pages/70503475/Bulk+Processing+with+GPT). But I don't think the `processing_gpt` plugin supports that.

But you can come very close to that by putting formatting fields into the XML file, reading that file and replacing all the fields with the settings you want:

So if you have a file `resample-s2.xml` that looks like this:

```xml
<graph id="Resample_Sentinel-2">
  <version>1.0</version>
  <node id="resample-s2">
    <operator>Resample</operator>
      <sources>
        <sourceProduct>${sourceProduct}</sourceProduct>
      </sources>
      <parameters>
        <targetResolution>{resolution}</targetResolution>
        <upsampling>{up}</upsampling>
        <downsampling>{down}</downsampling>
        <flagDownsampling>{flag}</flagDownsampling>
        <resampleOnPyramidLevels>false</resampleOnPyramidLevels>
      </parameters> 
   </node>
</graph>
```

You treat it like this:

```
import os

TEMPLATE_PATH = os.path.join(os.path.dirname(__file__), 'gpt_templates', 'resample-s2.xml')

with open(TEMPLATE_PATH) as f:
    template_fmt = f.read()

template_filled = template_fmt.format(
    resolution=20,
    down='Min',
    up='Bicubic',
    flag='First',
)

output_graph_file = '/path/to/my/graph-file.xml'
with open(output_graph_file, 'w') as f:
    f.write(template_filled)
```

Reply:
The nice thing about this solution compared to yours is that you actually have an XML file to work with - to make changes, add, remove lines, etc. So please give it a shot implementing this.

Reply:
I realized that there are some unused lines in our code which I will remove in the next days.

Reply:
Thanks for the response. I see that your motivation for this design is not filling in the variables but to have composable blocks. I still think that a bunch of static XML files would be cleaner, more readable and easier to maintain. But since you are the one maintaining this anyways, let's keep it as it is. 😄 Please open a new PR for the cleanup.